### DISCUSSION TITLE
Not able to deploy foundry lottery to real testnet

### DISCUSSION THREAD

**Discussion by liumOazed** [Cyfrin/foundry-full-course-cu#2800] [Q&A] [answered]
I am at the end of Foundry-Lottery. I am trying the following command 
```
forge script script/DeployRaffle.s.sol:DeployRaffle --rpc-url $(SEPOLIA_RPC_URL) --account default --broadcast --verify --etherscan-api-key $(ETHERSCAN_API_KEY) -vvvv
```

But  I am getting the following error:
```
SEPOLIA_RPC_URL: command not found
ETHERSCAN_API_KEY: command not found
error: a value is required for '--fork-url <URL>' but none was supplied
```

I have the following in my ```.env``` file:

```
SEPOLIA_RPC_URL=https://eth-sepolia.g.alchemy.com/v2/AXYMSmwqEhvfbkSwyRAVBao4QwbY7gaN
MAINNET_RPC_URL=https://eth-mainnet.g.alchemy.com/v2/tiDmjcNv7pd6qSYE0ev4q0TLrMMmwEyx
ETHERSCAN_API_KEY=3GT42UEDRDMBWG7HHSQA8F9SHH3X5X4WNG
```

I have done ```source .env```

But still getting the persistent error. 

```default``` is the private key of my metamask account that has link in it.

I am adding the screenshots as follows:
![image](https://github.com/user-attachments/assets/3bfe4a19-97cb-47d5-a565-7025189622b0)
![image](https://github.com/user-attachments/assets/5715e61f-f3e4-476c-9009-d2d17cc509bc)
![image](https://github.com/user-attachments/assets/558b5d93-898d-4a3d-a171-652d5d3cb1ee)

**Comment by EngrPips** [comment]
Is the `.env` file in the root folder of your working directory?

**Reply by liumOazed** [reply]
Yes.
![image](https://github.com/user-attachments/assets/5d3fdd7c-f37f-4e7b-ac7e-8f39554b9ff6)

**Reply by EngrPips** [reply]
Ensure you don't get into the habit of publicly showing your `.env` file like this. It appears to me that you have done everything correctly, and it is supposed to work. I am unsure what is wrong here, but I trust some giga Chad will spot the issue and help as soon as possible. Are you using wsl?

**Reply by liumOazed** [reply]
I understand your concern about showing .env variables. And yes I am using wsl.

**Comment by heheboii11** [comment]
Hey @liumOazed, can you just rerun the command in the terminal by removing (). The command should look like this 
```forge script script/DeployRaffle.s.sol:DeployRaffle --rpc-url $SEPOLIA_RPC_URL --account default --broadcast --verify --etherscan-api-key $ETHERSCAN_API_KEY -vvvv```
Please let me know if you still face the issue after trying this

**Reply by EngrPips** [reply]
nice catch.

**Reply by liumOazed** [reply]
@heheboii11 I ran without the parenthesis (). But existing error persists.
 
![image](https://github.com/user-attachments/assets/c6f260b2-74ea-47f3-8205-f08063c836f0)

I tried to run the same thing using ```Makefile``` but nothing happens. 
![image](https://github.com/user-attachments/assets/29d631b0-5fac-4755-bfac-c5c84326317b)

My test runs all okay locally. And I have sent LINK to the consumer using the ```default``` key as well. It's just the ```.env``` error persisting at the moment and I can't move on.
![image](https://github.com/user-attachments/assets/cb7939c8-95bf-4e63-845d-3606307c2c9e)

**Reply by EngrPips** [reply]
You should close and re-open `vscode`, run `forge clean`, delete your `cache` folder if there is any in your codebase, then make sure your `.env` file has all the variables you need to run your process. Then run `source .env` and retry. I hope it works.

**Reply by liumOazed** [reply, marked as answer]
Hey @EngrPips @heheboii11 I figured it out! ``` function run()``` was previously empty inside my ```DeployRaffle.s.sol``` I just fixed it and it worked! Thank you guys for your support nonetheless. Cheers!
![image](https://github.com/user-attachments/assets/a801fd3f-7c8a-4add-bb12-9d7f79997d44)

**Reply by EngrPips** [reply]
I am glad you figured it out. Keep crushing it.

**Comment by heheboii11** [comment]
My brain isn't braining 😅, lol!!. Do ```source .env``` once again and redo everything. or try using ```fork-url``` instead of ```rpc-url``` . If nothing works, I hope someone will help you out
